src/main/webapp/WEB-INF/message-thread.jsp

```diff
@@ -0,0 +1,78 @@
+<%@ page contentType="text/html" pageEncoding="UTF-8"%>
+<%@ page import="java.util.Date" %>
+<%@ page import="java.util.*" %>
+<%@ page import="com.google.codeu.data.Message" %>
+<%@ page import="com.google.codeu.data.Reply" %>
+<%@ page import="com.google.appengine.api.blobstore.BlobstoreService" %>
+<%@ page import="com.google.appengine.api.blobstore.BlobstoreServiceFactory" %>
+
+
+<% Message parent = (Message) request.getAttribute("parent"); %>
+<% List<Reply> replies = (List<Reply>) request.getAttribute("replies"); %>
+<% String currentUser = (String) request.getAttribute("currentUser"); %>
+
+
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>Message Thread</title>
+    <meta charset="UTF-8">
+    <link rel="stylesheet" href="/css/main.css">
+    <link rel="stylesheet" href="/css/user-page.css">
+    <script src="/js/message-loader.js"></script>
+    <script src="https://cdn.ckeditor.com/ckeditor5/11.2.0/classic/ckeditor.js"></script>
+  </head>
+  <body onload="buildUI()">
+    <div class="navbar">
+      <a href="/">Home</a>
+    <% if (currentUser != null) { %>
+      <a href="/users/<%=currentUser%>">Your Page</a>
+    <% } else { %>
+      <a href="/login">Login</a>
+    <% } %>
+    </div>
+    <h1 id="page-title"> Message Thread </h1>
+    <% if(currentUser != null){ %>
+    <% String parentID = parent.getId().toString(); %>
+      <form id="reply-form" action="/reply" method="POST">
+        Add a reply:
+        <br/>
+        <textarea name="text" placeholder="Enter a message" id="message-input"></textarea>
+        <input type="hidden" name="parentID" value="<%=parentID%>">
+        <br/>
+        <button type="submit" value="Submit"> SUBMIT </button>
+      </form>
+    <% }  %>
+
+    <div id="message-container">
+      <div class="message-div">
+        <div class="message-header">
+          User: <%= parent.getUser() %> --
+          Time: <%= new Date(parent.getTimestamp()) %> --
+          Country: <%= parent.getCountry() %> --
+          Category: <%= parent.getCategory() %>
+        </div>
+        <div class="message-body">
+        <% if(parent.hasAnImage()){ %>
+          <%= parent.getText() + "<br/>" + "<img src=\"" + parent.getImageUrl() + "\"/>"%>
+        <% } else { %>
+          <%= parent.getText() %>
+        <% } %>
+        </div>
+      </div>
+      <% for(int i = 0; i < replies.size(); i++) { %>
+          <div class="reply-div">
+            <div class="reply-header">
+              User: <%= replies.get(i).getUser() %> --
+              Time: <%= new Date(replies.get(i).getTimestamp()) %>
+            </div>
+            <div class="reply-body">
```

Comment:
Did you change the div classes from message to reply for any reason? Unless you were planning to style the replies differently than other messages, you can make them uniform by leaving it as message-div, message-header, and message-body, that why they're all styled the same way from main.css. But if you wanted to leave the boxes out that are normally around messages, that's fine too!

Reply:
Makes sense, I'll change those back. I was wondering why that was